src/content/tools/hooks.md

```diff
@@ -0,0 +1,377 @@
+    ```
+
+    * `input`: The input for `hook/build.dart`. This
+      includes the target OS, target architecture,
+      output directory, etc. For details, see
+      the [`BuildInput`][] class.
+
+    * `output`: The output for `hook/build.dart`. This
+      includes the assets to be bundled, and dependencies.
+      For details, see the [`BuildOutputBuilder`][] class.
+
+[`BuildInput`]: {{site.pub-api}}/hooks/latest/hooks/BuildInput-class.html
+[`BuildOutputBuilder`]: {{site.pub-api}}/hooks/latest/hooks/BuildOutputBuilder-class.html
+
+### Create a link hook script
+
+If you want to use a link hook, follow these steps to
+add one to a Dart script: 
+
+1.  In your Dart project, create or open `hooks/link.dart`. 
+
+1.  In the main method, add a `link` function from
+    `package:hooks/hooks.dart` and construct a data asset.
+    For example:
+
+    ```dart title="hooks/link.dart" highlightLines=14
+    import 'dart:convert';
+    import 'dart:io';
+
+    import 'package:data_assets/data_assets.dart';
+    import 'package:hooks/hooks.dart';
+    import 'package:record_use/record_use.dart';
+
+    const multiplyIdentifier = Identifier(
+      importUri: 'package:package_with_assets/package_with_assets.dart',
+      name: 'AssetUsed',
+    );
+
+    void main(List<String> args) async {
+      await link(args, (input, output) async {
+        final usages = input.usages;
+
+        final usedAssets = (usages.instancesOf(multiplyIdentifier) ?? []).map(
+          (e) => (e.instanceConstant.fields.values.first as StringConstant).value,
+        );
+
+        output.assets.data.addAll(
+          input.assets.data.where(
+            (dataAsset) => usedAssets.contains(dataAsset.name),
+          ),
+        );
+      });
+    }
+
+    extension on LinkInput {
+      RecordedUsages get usages {
+        final usagesFile = recordedUsagesFile;
+        final usagesContent = File.fromUri(usagesFile!).readAsStringSync();
+        final usagesJson = jsonDecode(usagesContent) as Map<String, Object?>;
+        final usages = RecordedUsages.fromJson(usagesJson);
+        return usages;
+      }
+    }
+    ```
+
+    * `input`: The input for `hook/link.dart`. This
+      includes the target OS, target architecture,
+      output directory, etc. For details, see
+      the [`LinkInput`][] class.
+
+    * `output`: The output for `hook/link.dart`. This
+      includes the assets to be bundled, and dependencies.
+      For details, see the [`LinkOutputBuilder`][] class.
+
+[`LinkInput`]: {{site.pub-api}}/hooks/latest/hooks/LinkInput-class.html
+[`LinkOutputBuilder`]: {{site.pub-api}}/hooks/latest/hooks/LinkOutputBuilder-class.html
+
+### Generate assets
+
+Assets are generated when you invoke the Dart hooks in
+the `hook/` directory. Your Dart hooks are invoked when when
+you use the `run`, `build`, or `test` command on your
+package.
+
+### Reference assets
+
+Assets are the files that hooks create in the
+`build/` directory. Once an asset has been created,
```

Comment:
This is not true. It might be true for flutter, but not for Dart.

`dart build` creates a `build/` directory.

The assets are actually generated in `.dart_tool/build_runner/<...>/`, but I'm
not sure how much of that we want to tell users. They can see this from
inspecting `input.outputDirectoryShared`.

Reply:
Cool. I've updated the docs.